pagerduty/handler.py

```diff
@@ -112,41 +110,66 @@ def proxy_settings(self, settings=None, event=None):
             proxy_settings['proxy_password'] = ''
         return proxy_settings
 
-    def contexts(self, settings=None, event=None):
-        if not settings:
-            settings = self.settings
-        if not event:
-            event = self.event
+    def contexts(self):
```

Comment:
Contexts too big changes to enable the ability for checks to provide hashes, arrays or strings as output. 
We did this so we can clean up custom_details in the actual incident payload to be more clear what is happening as well as giving me a way to put any check hook output, descriptive text or other helpful info into the incident.